etc/helm/pachyderm/templates/pachd/deployment.yaml

```diff
@@ -55,7 +55,7 @@ spec:
         env:
         - name: GOCDK_ENABLED
           value: {{ .Values.pachd.storage.gocdkEnabled | default true | quote }}
-        {{- if .Values.pachd.storage.gocdkEnabled }}
+        {{- if and (ne .Values.deployTarget "LOCAL") .Values.pachd.storage.gocdkEnabled  }}
```

Comment:
When the deploy target is local, we just use the `storageRoot` as the object storage backend